### ISSUE TITLE
Cannot acquire exclusive lock to create or update server farm.

### ISSUE THREAD

**Issue by gunzip** [terraform-providers/terraform-provider-azurerm#600]
Hi there,

### Terraform Version

```
$ terraform.exe -v
Terraform v0.10.6
```

### Affected Resource(s)

-  app_service_plan

### Terraform Configuration Files

Terraform resources configuration:
https://github.com/gunzip/digital-citizenship/blob/apim-conf-db/infrastructure/azure.tf#L241

Variables:
https://github.com/gunzip/digital-citizenship/blob/apim-conf-db/infrastructure/env/test/tfvars.json

### Debug Output

Resource: /subscriptions/eXXXXXX-cXXX-4XXX-bXXX-dXXXXXX/resourceGroups/agid-rg-test/providers/Microsoft.Web/serverfarms/agid-app-test

Output running terraform:
```
Error applying plan:

1 error(s) occurred:

* azurerm_app_service_plan.azurerm_app_service_plan: 1 error(s) occurred:

* azurerm_app_service_plan.azurerm_app_service_plan: web.AppServicePlansClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Put https://management.azure.com/subscriptions/eXXXXXX-cXXX-4XXX-bXXX-dXXXXXX/resourceGroups/agid-rg-test/providers/Microsoft.Web/serverfarms/agid-app-test?api-version=2016-09-01: http: ContentLength=131 with Body length 0
```

From Azure resource group logs:
```
Cannot acquire exclusive lock to create or update this server farm. Retry the request later.`
```

### Expected Behavior

The resource should have been created.

### Actual Behavior

Terraform exits with errors.

### Steps to Reproduce

Run `terraform apply`
with the provided configuration.

### Important Factoids

The error goes away running 'terraform apply' twice.

**Comment by tombuildsstuff** [comment]
hey @gunzip 

Thanks for opening this issue :)

So that we can take a further look into this and reach out to the App Service Team - would it be possible to know which HTTP Status Code is returned along with the `Cannot acquire exclusive lock to create or update this server farm. Retry the request later.` error?
If you've still got access to the logs then this should be available near the original error - otherwise we can attempt to reproduce this, however it's historically harder to reproduce these kinds of errors

Thanks!

**Comment by gunzip** [comment]
hi @tombuildsstuff, this is a gist with the full error trace:

https://gist.github.com/gunzip/46e32eedaaedf57902195e8a4ec634c9

Code returned is 429 (maybe a conflict creating two app service plans together ?)

**Comment by tombuildsstuff** [comment]
Hey @gunzip 

Thanks for that - that's really useful.

Taking a look into the log you've provided this appears to be a bug in the Azure SDK for Go - where the status code 429 isn't being polled for - as such [I've opened an issue about this on the Go SDK Repository](https://github.com/Azure/azure-sdk-for-go/issues/884) to request this status code is polled for.

Thanks!

**Comment by tombuildsstuff** [comment]
Hey @gunzip 

The issue's been fixed in Go-AutoRest (which is used by the Go SDK) and has been released in v1.0.0 of the AzureRM Terraform Provider. Would it be possible to take a look and see if this solves your issue?

I'm going to close this issue in the interim since I believe this is fixed (but please feel free to re-open it if it's still an issue and we'll take another look).

Thanks!
